src/flb_log.c

```diff
@@ -57,10 +57,82 @@ static inline int consume_byte(flb_pipefd_t fd)
     return 0;
 }
 
+static inline int log_file_sz_max(off_t size, struct flb_log *log)
+{
+    double sz_mb;
+
+    sz_mb = size / (1024.00 * 1024.00);
+    if (sz_mb >= log->max_sz_mb) {
+        return FLB_TRUE;
+    }
+
+    return FLB_FALSE;
+}
+
+static inline void log_file_rotate(struct flb_log *log)
+{
+    char *backup_file_name = NULL;
+    char *src_file_name = NULL;
+    int ret = -1;
+
+    if (log->max_history == 0) {
+        /*
+         * no backups.
+         * remove the file since size has exceeded.
+         */
+        remove(log->out);
+        return;
+    }
+
+    /* max 2 digits for backup name suffix and 1 each for '.' and '\0' */
+    size_t filename_len = strlen(log->out) + 4;
+    backup_file_name = flb_malloc(filename_len);
+    if (!backup_file_name) {
+        flb_errno();
+        return;
+    }
+    src_file_name = flb_malloc(filename_len);
+    if (!src_file_name) {
+        flb_errno();
+        return;
+    }
+
+    int l;
+    for (l = log->max_history; l > 1; l--) {
+        snprintf(backup_file_name, filename_len, "%s.%d", log->out, l);
+        snprintf(src_file_name, filename_len, "%s.%d", log->out, l-1);
+        /* since rename behaviour is implementation dependent when new file exists, remove the last file. */
+        if (l == log->max_history) {
+            remove(backup_file_name);
```

Comment:
we might need a Windows compat version for this. cc: @leonardo-albertovich

Reply:
remove is from stdio.h and hence shouldn't be an issue I believe.

Reply:
It is according to MSDN but I couldn't find / verify when was it introduced (Sorry windows 95 users!)